controllers/provisioning_controller.go

```diff
@@ -650,14 +654,45 @@ func (r *ProvisioningReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	klog.InfoS("Network stack calculation", "NetworkStack", r.NetworkStack)
 
+	provisioningFilter := predicate.NewPredicateFuncs(func(object client.Object) bool {
+		if object.GetName() == metal3iov1alpha1.ProvisioningSingletonName {
+			return true
+		}
+		klog.Info("ignoring provisioning config with unexpected name", "name", object.GetName(), "expected-name", metal3iov1alpha1.ProvisioningSingletonName)
+		return false
+	})
+
+	// Watch Secret openshift-config/pull-secret. If this secret changes, we must requeue the provisioning Singleton,
+	// if it exists.
+	watchOCPConfigPullSecret := func(ctx context.Context, object client.Object) []reconcile.Request {
```

Comment:
nit: this variable is confusingly named: it's not used just for the pull secret

Reply:
Agree, and it doesn't do a watch either.
We should call it something like `mapToProvisioningSingleton`.